GoogleUtilities/Network/GULNetworkURLSession.m

```diff
@@ -157,7 +157,15 @@ - (nullable NSString *)sessionIDFromAsyncPOSTRequest:(NSURLRequest *)request
     session = [NSURLSession sessionWithConfiguration:_sessionConfig
                                             delegate:self
                                        delegateQueue:[NSOperationQueue mainQueue]];
-    postRequestTask = [session uploadTaskWithRequest:request fromData:request.HTTPBody];
+    // To avoid a runtime warning in Xcode 15 Beta 4, the given `URLRequest`
+    // should have a nil `HTTPBody`. To workaround this, the given `URLRequest`
+    // is copied and the `HTTPBody` data is removed.
+    NSData *givenRequestHTTPBody = [request.HTTPBody copy];
+    NSMutableURLRequest *requestWithoutHTTPBody = [request mutableCopy];
+    requestWithoutHTTPBody.HTTPBody = nil;
+
+    postRequestTask = [session uploadTaskWithRequest:requestWithoutHTTPBody.copy
+                                            fromData:givenRequestHTTPBody];
```

Comment:
Seems like there are two options, do this or add/change the public API to take a data-less request (enforced through an assert?) and a data parameter. I think this PR's current implementation is preferred.

Reply:
Agreed